### ISSUE TITLE
codecov doesn't work for pull request except created by me

### ISSUE THREAD

**Issue by redboltz** [redboltz/mqtt_cpp#584]
See https://github.com/msgpack/msgpack-c/issues/832

codecov will provide uploading coverage data without token like travis-ci. In order to do that, codecov need to use github actions APIs. However, the APIs are not published yet. According to github, it will be published early this year.

**Comment by redboltz** [comment]
Why except me, it is because I only can access github secret. So my pull request can upload coverage data to codecov.

I don't have strong motivation to support codecov because it will be supported by github and codecov soon.
Our possible options are hard code token string, use travis-ci for codecov, or just wait.
I'd like to wait.

**Comment by redboltz** [comment]
I studied little more about codecov. And I started thinking hard-coded token could be am acceptable temporary solution.
If I add hard-coded token to 
https://github.com/redboltz/mqtt_cpp/blob/d0049c37b793e918bfe06bfce3db17fcb455a7ce/.github/workflows/gha.yml#L171
codecov for pull request works well.

Public codecov uploading token means anyone can upload the coverage data. It is not good but I think that the bad effect is limited. In addition, if someone frequently upload the invalid coverage data, I can re-generate the token and stop this temporary workaround.
I think that it is acceptable temporary workaround. 

Any ideas?
